src/main/java/es/thalesalv/gptbot/application/service/commands/CreateLorebookEntryService.java

```diff
@@ -0,0 +1,133 @@
+    private final LorebookRegexRepository lorebookRegexRepository;
+    private final LorebookEntryToDTOTranslator lorebookEntryToDTOTranslator;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CreateLorebookEntryService.class);
+    private static final String LORE_ENTRY_CREATED = "Lore entry with name **{0}** created. Don't forget to save this ID!\n```json\n{1}\n```";
+
+    @Override
+    public void handle(final SlashCommandInteractionEvent event) {
+
+        LOGGER.debug("Showing modal for character creation");
+        final Modal modal = buildEntryCreationModal();
+        event.replyModal(modal).queue();
+    }
+
+    @Override
+    public void handle(final ModalInteractionEvent event) {
+
+        try {
+            LOGGER.debug("Received data from character creation modal -> {}", event.getValues());
+            event.deferReply();
+
+            final User author = event.getMember().getUser();
+            final String entryName = event.getValue("lorebook-entry-name").getAsString();
+            final String entryRegex = event.getValue("lorebook-entry-regex").getAsString();
+            final String entryDescription = event.getValue("lorebook-entry-desc").getAsString();
+            final String entryPlayerCharacter = event.getValue("lorebook-entry-player").getAsString();
+            final boolean isPlayerCharacter = entryPlayerCharacter.equals("y");
+            final UUID lorebookEntryId = Generators.randomBasedGenerator().generate();
+            final UUID lorebookRegexId = Generators.randomBasedGenerator().generate();
+            final LorebookRegex insertedEntry = insertEntry(author, entryName, entryRegex,
+                    entryDescription, lorebookEntryId, lorebookRegexId, isPlayerCharacter);
+
+            final LorebookDTO loreItem = lorebookEntryToDTOTranslator.apply(insertedEntry);
+            final String loreEntryJson = objectMapper.setSerializationInclusion(Include.NON_NULL)
+                    .writerWithDefaultPrettyPrinter().writeValueAsString(loreItem);
+
+            event.reply(MessageFormat.format(LORE_ENTRY_CREATED,
+                            insertedEntry.getLorebookEntry().getName(), loreEntryJson))
+                    .setEphemeral(true).complete();
+        } catch (Exception e) {
+            LOGGER.error("Error processing entry object", e);
+        }
+    }
+
+    private Modal buildEntryCreationModal() {
+
+        LOGGER.debug("Building entry creation modal");
+        final TextInput lorebookEntryName = TextInput
+                .create("lorebook-entry-name", "Name", TextInputStyle.SHORT)
+                .setPlaceholder("Forest of the Talking Trees")
+                .setRequired(true)
+                .build();
+
+        final TextInput lorebookEntryRegex = TextInput
+                .create("lorebook-entry-regex", "Regular Expression (optional)", TextInputStyle.SHORT)
+                .setPlaceholder("/(Rain|)Forest of the (Talking|Speaking) Trees/gi")
```

Comment:
FYI, `(Rain|)` could just as easily be `(Rain)?` Either is fine. I think the latter is more conventional, but yours is a valid choice.

Reply:
(Rain|) matches both "Rain" and empty, while (Rain) is just a group that matches only that specific word, they have different results. Still, that's just a placeholedr that serves as example for the user

Reply:
`(Rain)?forest` also matches both "rainforest" and "forest" and is the more conventional way of doing it.

Reply:
But I do want to emphasize there is absolutely nothing wrong with your way. It's just weird to me as someone who occasionally uses regex.